crates/parcel_plugin_transformer_js/src/transformer.rs

```diff
@@ -1,24 +1,894 @@
+      None,
+    )?;
+
+    // TODO handle errors properly
+    if let Some(errors) = transformation_result.diagnostics {
+      return Err(anyhow!(format!("{:#?}", errors)));
+    }
+
+    let asset = Asset::new_empty(input.file_path().to_path_buf(), source_code);
+    let config = parcel_js_swc_core::Config::default();
+    let options = ParcelOptions::default();
+    let result = convert_result(asset, &config, transformation_result, &options)
+      // TODO handle errors properly
+      .map_err(|_err| anyhow!("Failed to transform"))?;
+
+    Ok(result)
+  }
+}
+
+fn convert_result(
+  mut asset: Asset,
+  transformer_config: &parcel_js_swc_core::Config,
+  result: parcel_js_swc_core::TransformResult,
+  options: &ParcelOptions,
+) -> Result<TransformResult, Vec<Diagnostic>> {
+  let asset_file_path = asset.file_path.to_path_buf();
+  let asset_environment = asset.env.clone();
+  let asset_id = asset.id();
+
+  if let Some(shebang) = result.shebang {
+    asset.set_interpreter(shebang);
+  }
+
+  let (mut dependency_by_specifier, invalidate_on_file_change) = convert_dependencies(
+    transformer_config,
+    result.dependencies,
+    &asset_file_path,
+    &asset_environment,
+    asset_id,
+  )?;
+
+  if result.needs_esm_helpers {
+    let mut dependency_flags = DependencyFlags::empty();
+    dependency_flags.set(
+      DependencyFlags::HAS_SYMBOLS,
+      result.hoist_result.is_some() || result.symbol_result.is_some(),
+    );
+    let dependency = make_esm_helpers_dependency(
+      options,
+      &asset_file_path,
+      asset_environment,
+      dependency_flags,
+      asset_id,
+    );
+    dependency_by_specifier.insert(dependency.specifier.as_str().into(), dependency);
+  }
+
+  if let Some(hoist_result) = result.hoist_result {
+    asset.flags.set(AssetFlags::HAS_SYMBOLS, true);
+    // Pre-allocate expected symbols
+    asset
+      .symbols
+      .reserve(hoist_result.exported_symbols.len() + hoist_result.re_exports.len() + 1);
+
+    // Collect all exported variable names into `asset.symbols`
+    for symbol in &hoist_result.exported_symbols {
+      let symbol = transformer_exported_symbol_into_symbol(&asset_file_path, &symbol);
+      asset.symbols.push(symbol);
+    }
+
+    // Collect all imported symbols into each of the corresponding dependencies' symbols array
+    for symbol in hoist_result.imported_symbols {
+      if let Some(dependency) = dependency_by_specifier.get_mut(&symbol.source) {
+        let symbol = transformer_imported_symbol_to_symbol(&asset_file_path, &symbol);
+        dependency.symbols.push(symbol);
+      }
+    }
+
+    for symbol in hoist_result.re_exports {
+      if let Some(dependency) = dependency_by_specifier.get_mut(&symbol.source) {
+        if is_re_export_all_symbol(&symbol) {
+          let loc = Some(convert_loc(asset_file_path.clone(), &symbol.loc));
+          dependency.symbols.push(make_export_all_symbol(loc));
+          // TODO: Why isn't this added to the asset.symbols array?
```

Comment:
Why isn't this added to the `asset.symbols` array?

Reply:
This represents an `export * from '...'` statement. There can be multiple of those in the same file so they need to be on each dependency rather than on the asset.

The representation of symbols really needs to change though tbh. At the moment this is mostly done for compatibility with the representation in JS. There, we actually use a `Map` rather than an array for symbols so there cannot be multiple symbols of the same name. That eventually needs to change.

Reply:
Wouldn't all re-exports have the same issue? Is it easier to just collect on both vectors so we get consistent outputs?

Reply:
You could try changing the JS implementation here and running the tests to see what breaks: https://github.com/parcel-bundler/parcel/blob/c03a29ede215967a9ec4e472a1161eed0a913af6/packages/transformers/js/src/JSTransformer.js#L905

I suspect it's because there could be both an `export *` dependency _and_ a namespace (`*`) symbol defined within the asset. e.g.

```js
export * from './foo.js';

exports[someNonStaticName] = 2;
```

In this case, the module is not statically analyzable, so it has a `*` namespace symbol in the asset. However it also exports * from foo.js. Since the JS implementation uses a map to represent symbols, these would conflict if they both defined the `*` symbol on the asset.

We can probably change the representation of symbols later to fix this but for now I went with compatibility with the JS implementation (which we need for the packager etc.)